backvendor/gosource.go

```diff
@@ -346,6 +346,11 @@ func importPathFromFilepath(path string) (string, bool) {
 	}
 
 	for i := len(components) - 2; i >= 0; i -= 1 {
+		switch components[i] {
+		case "testdata", "vendor":
```

Comment:
It's not clear to me when you'd actually encounter this. Is this if backvendor is run against a vendored package directly but the package can't be deduced from the path and you want to give up once you hit the "vendor" directory?

Reply:
I added a comment, and also pushed the commits separately to show the test results at each stage.

Essentially, adding a simple test case which expected not to find an import comment in a package inside "testdata" actually found an import path for backvendor itself, because it kept searching even outside "testdata".